nexus/src/app/silo.rs

```diff
@@ -383,35 +383,38 @@ impl super::Nexus {
             )
             .await?;
 
-        let (authz_silo_user, db_silo_user) =
-            if let Some(existing_silo_user) = fetch_result {
-                existing_silo_user
-            } else {
-                // In this branch, no user exists for the authenticated subject
-                // external id. The next action depends on the silo's user
-                // provision type.
-                match db_silo.user_provision_type {
-                    // If the user provision type is ApiOnly, do not create a
-                    // new user if one does not exist.
-                    db::model::UserProvisionType::ApiOnly => {
-                        return Ok(None);
-                    }
+        let (authz_silo_user, db_silo_user) = if let Some(existing_silo_user) =
+            fetch_result
+        {
+            existing_silo_user
+        } else {
+            // In this branch, no user exists for the authenticated subject
+            // external id. The next action depends on the silo's user
+            // provision type.
+            match db_silo.user_provision_type {
+                // If the user provision type is ApiOnly, do not create a
+                // new user if one does not exist.
+                db::model::UserProvisionType::ApiOnly => {
+                    return Err(Error::Unauthenticated {
+                            internal_message: "User must exist before login when user provision type is ApiOnly".to_string(),
+                    });
```

Comment:
This diff is annoying due to indentation changes, but the only actual change here was changing `return Ok(None);` to this.

Reply:
I'm pretty confident but I would appreciate a very quick look @jmpesp since you wrote this way back in #1358.